### PR TITLE
Convert wdqs docker image to not run blazegraph as root

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: T237248

### CODE REVIEW

File: wdqs/0.3.10/Dockerfile [hunk 1]
```diff
@@ -15,10 +15,12 @@ FROM openjdk:8-jdk-alpine
 # Install gettext for envsubst command, (it needs libintl package)
 # Install curl for the loadData.sh wdqs script (if someone needs it)
 RUN set -x ; \
-    apk --no-cache add bash=\<4.5.0 gettext=\<0.19.8.2 libintl=\<0.19.8.2 curl=\<7.64.999
+    apk --no-cache add bash=\<4.5.0 gettext=\<0.19.8.2 libintl=\<0.19.8.2 curl=\<7.64.999 sudo=\<1.9
```
Comment (tarrow) [inline wdqs/0.3.10/Dockerfile]: Is it possible to just use plan `su` so we don't need to add this dependency? I don't think blazegraph needs to be on an interactive TTY so maybe we could just use `su -c` ?

Comment (Ladsgroup) [reply wdqs/0.3.10/Dockerfile]: su is not super great in normal computer use but I understand this is different in containers, let me fix it then.

Comment (Ladsgroup) [reply wdqs/0.3.10/Dockerfile]: su doesn't work in this case, "runuser" might.

Comment (addshore) [reply wdqs/0.3.10/Dockerfile]: > su doesn't work in this case

What's the problem?
 sudo -u doesnt work? :(

Maybe we could / should do something like https://github.com/elastic/dockerfiles/blob/v7.5.1/elasticsearch/bin/docker-entrypoint.sh#L7-L16

Comment (Ladsgroup) [reply wdqs/0.3.10/Dockerfile]: sudo -u needs "sudo" package to be installed in the alpine docker images

**Comment by wiese** [comment]
It's been a while but has the [`USER`](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user) command been considered when this was implemented?

**Comment by Ladsgroup** [comment]
> It's been a while but has the [`USER`](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user) ~command~ instruction been considered when this was implemented?

I'm pretty sure I looked at it and it didn't work but I don't remember why. Give me a couple of hours, I will tell you.

**Comment by wiese** [comment]
> Give me a couple of hours, I will tell you.

It does work, there is clearly no rush. Was merely interested what spoke against it ("Avoid installing or using sudo...").

**Comment by Ladsgroup** [comment]
Found it: https://phabricator.wikimedia.org/T237248#5744506

The USER can't change permissions on root and thus the container can't write to the mounted volumes (I tried it in https://github.com/wmde/wikibase-docker/commit/a8e3c9c9e6abcb3c46c9775d320d4c010750002f)

**Comment by addshore** [comment]
So we just ran into this while updating the wikibase-registry. :/ looking into it now.

https://github.com/wmde/wikibase-docker/commit/e38eaf29456f2a83d51dab02a06d262c49d1541c probably wont work due to the pre existing permissions on the file that need to be changed in the entry point.

**Comment by wiese** [comment]
> e38eaf2 probably wont work due to the pre existing permissions

Would it not make sense to differentiate between one time fixes and general system behavior?

This again touches on the topic of using containers as "virtual machines".

**Pull request merged**